core/drivers/stm32_tamp.c

```diff
@@ -93,12 +96,103 @@
 #define _TAMP_PRIVCFGR_MASK		(GENMASK_32(31, 29) | \
 					 GENMASK_32(15, 14))
 
+/*_TAMP_CR1 bit fields */
+#define _TAMP_CR1_ITAMP(id)		BIT((id) - INT_TAMP1 + U(16))
+#define _TAMP_CR1_ETAMP(id)		BIT((id) - EXT_TAMP1)
+
+/* _TAMP_CR2 bit fields */
+#define _TAMP_CR2_ETAMPTRG(id)		BIT((id) - EXT_TAMP1 + U(24))
+#define _TAMP_CR2_BKERASE		BIT(23)
+#define _TAMP_CR2_BKBLOCK		BIT(22)
+#define _TAMP_CR2_ETAMPMSK_MAX_ID	3U
+#define _TAMP_CR2_ETAMPMSK(id)		BIT((id) - EXT_TAMP1 + U(16))
+#define _TAMP_CR2_ETAMPNOER(id)		BIT((id) - EXT_TAMP1)
+
+/* _TAMP_CR3 bit fields */
+#define _TAMP_CR3_ITAMPNOER_ALL		GENMASK_32(12, 0)
+#define _TAMP_CR3_ITAMPNOER(id)		BIT((id) - INT_TAMP1)
+
+/* _TAMP_FLTCR bit fields */
+#define _TAMP_FLTCR_TAMPFREQ_MASK	GENMASK_32(2, 0)
+#define _TAMP_FLTCR_TAMPFREQ_SHIFT	0U
+#define _TAMP_FLTCR_TAMPFLT_MASK	GENMASK_32(4, 3)
+#define _TAMP_FLTCR_TAMPFLT_SHIFT	U(3)
+#define _TAMP_FLTCR_TAMPPRCH_MASK	GENMASK_32(6, 5)
+#define _TAMP_FLTCR_TAMPPRCH_SHIFT	5U
+#define _TAMP_FLTCR_TAMPPUDIS		BIT(7)
+
+/* _TAMP_ATCR bit fields */
+#define _TAMP_ATCR1_ATCKSEL_MASK	GENMASK_32(19, 16)
+#define _TAMP_ATCR1_ATCKSEL_SHIFT	16U
+#define _TAMP_ATCR1_ATPER_MASK		GENMASK_32(26, 24)
+#define _TAMP_ATCR1_ATPER_SHIFT		24U
+#define _TAMP_ATCR1_ATOSHARE		BIT(30)
+#define _TAMP_ATCR1_FLTEN		BIT(31)
+#define _TAMP_ATCR1_COMMON_MASK		GENMASK_32(31, 16)
+#define _TAMP_ATCR1_ETAMPAM(id)		BIT((id) - EXT_TAMP1)
+#define _TAMP_ATCR1_ATOSEL_MASK(id) \
+	GENMASK_32(((id) - EXT_TAMP1 + 1) * 2 + 7, ((id) - EXT_TAMP1) * 2 + 8)
```

Comment:
```suggestion
#define _TAMP_ATCR1_ATOSEL_MASK(id) \
	({ \
		typeof(id) _id = (id); \
		GENMASK_32(((_id) - EXT_TAMP1 + 1) * 2 + 7, ((_id) - EXT_TAMP1) * 2 + 8); \
	})
```
...should get rid of the checkpatch warning (and make the macro safer to use)